src/check.h.in

```diff
@@ -669,6 +671,263 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
+#define _ck_assert_floating_op_prec(X, OP, Y, P, D) do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  int _ck_p = (P); \
+  ck_assert_floating_prec(P); \
+  ck_assert_msg(_ck_x - _ck_y OP powl(10, -_ck_p) * (D), \
+  "Assertion '%s' failed: %s == %.*Lg, %s == %.*Lg, %s == %d", \
+  #X" "#OP" "#Y" with precision 10^-"#P, \
+  #X, (int)CK_FLOATING_DIG, _ck_x, \
+  #Y, (int)CK_FLOATING_DIG, _ck_y, \
+  #P, _ck_p); \
+} while (0)
+
+/**
+ * Check two floating point numbers to determine if X>Y with specified precision
+ *
+ * If not X>Y with specified precision, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param PREC positive number of decimal places to be compared (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_gt_prec(X, Y, P) _ck_assert_floating_op_prec(X, >, Y, P, -1)
+
+/**
+ * Check two floating point numbers to determine if X>=Y with specified
+ * precision
+ *
+ * If not X>=Y with specified precision, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param PREC positive number of decimal places to be compared (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ge_prec(X, Y, P) _ck_assert_floating_op_prec(X, >=, Y, P, -1)
+
+/**
+ * Check two floating point numbers to determine if X<Y with specified precision
+ *
+ * If not X<Y with specified precision, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param PREC positive number of decimal places to be compared (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_lt_prec(X, Y, P) _ck_assert_floating_op_prec(X, <, Y, P, 1)
+
+/**
+ * Check two floating point numbers to determine if X<=Y with specified
+ * precision
+ *
+ * If not X<=Y with specified precision, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param PREC positive number of decimal places to be compared (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_le_prec(X, Y, P) _ck_assert_floating_op_prec(X, <=, Y, P, 1)
+
+/**
+ * Check two floating point numbers to determine if X≈Y with specified precision
+ *
+ * If not X≈Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param PREC positive number of decimal places to be compared (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_eq_prec(X, Y, PREC) \
```

Comment:
Could the code in ck_assert_floating_eq_prec and ck_assert_floating_ne_prec be combined and reused? It looks like the only difference is the operation used, either < or >=.